code/game/objects/items/toys.dm

```diff
@@ -1091,4 +1091,26 @@ obj/item/toy/cards/deck/syndicate
 	desc = "An adorable stuffed toy that resembles a space carp."
 	icon = 'icons/obj/toy.dmi'
 	icon_state = "carpplushie"
-	w_class = 2.0
\ No newline at end of file
+	w_class = 2.0
+
+/*
+ * Toy big red button
+ */
+/obj/item/toy/redbutton
+	name = "\improper Big Red Button toy"
+	desc = "A big, plastic red button. Who knows what it does?"
+	icon = 'icons/obj/assemblies.dmi'
+	icon_state = "bigred"
+	w_class = 2.0
+	var/cooldown = 0
+
+/obj/item/toy/redbutton/attack_self(mob/user)
+	if (cooldown < world.time)
+		cooldown = (world.time + 300) // Testing it at 30 seconds
+		user.visible_message("<span class='warning'>[user] presses the big red button.</span>", "<span class='notice'>You press the button, it plays a loud noise!</span>", "<span class='notice'>The button clicks loudly.</span>")
+		spawn(5) //gia said so
+			playsound(src, 'sound/effects/explosionfar.ogg', 50, 0, surround = 0)
+			sleep(135)
+			sleep(cooldown - world.time)

```

Comment:
We discussed this in IRC, for history's sake I aped the code from the nuke
and it had those in there. I removed the lines and I'll commit it shortly.

I'm also fiddling with a function to shake peoples' screens when it's used.
On Feb 3, 2015 3:21 AM, "Razharas" <notifications@github.com> wrote:

> In code/game/objects/items/toys.dm
> <https://github.com/tgstation/-tg-station/pull/7521#discussion_r23991159>:
>
> > +/obj/item/toy/redbutton
> > +	name = "\improper Big Red Button toy"
> > +	desc = "A big, plastic red button. Who knows what it does?"
> > +	icon = 'icons/obj/assemblies.dmi'
> > +	icon_state = "bigred"
> > +	w_class = 2.0
> > +	var/cooldown = 0
> > +
> > +/obj/item/toy/redbutton/attack_self(mob/user)
> > +	if (cooldown < world.time)
> > +		cooldown = (world.time + 300) // Testing it at 30 seconds
> > +		user.visible_message("<span class='warning'>[user] presses the big red button.</span>", "<span class='notice'>You press the button, it plays a loud noise!</span>", "<span class='notice'>The button clicks loudly.</span>")
> > +		spawn(5) //gia said so
> > +			playsound(src, 'sound/effects/explosionfar.ogg', 50, 0, surround = 0)
> > +			sleep(135)
> > +			sleep(cooldown - world.time)
>
> What is the purpose of these two lines?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/7521/files#r23991159>.
>